### PR TITLE
Webhooks

### PR DESCRIPTION
**Pull request by SnazzyPine25**
- [x] `Server#webhooks`
- [x] `Channel#webhooks`
- [x] `Channel#create_webhook` 
- [x] `Webhook#name=` & `Webhook#avatar=` just fails
- [x] `Webhook#delete`
- I know i'm screwing something up somehow

### CODE REVIEW

File: lib/discordrb/data.rb [hunk 1]
```diff
@@ -243,6 +243,7 @@ class Application
     # ID and avatar if the bot cannot reach the owner.
     # @return [User] the user object of the owner
     attr_reader :owner
+    alias_method :user, :owner
```
Comment (meew0) [inline lib/discordrb/data.rb]: Why is this here..?

File: lib/discordrb/data.rb [hunk 2]
```diff
@@ -269,6 +270,92 @@ def inspect
...
+      @name = data['name']
+      @id = data['id'].to_i
+      @channel = server.text_channels.find {|c| c.id == data['channel'].to_i}
+      @server = server
+      @token = data['token']
+      @avatar_id = data['avatar']
+      @owner = @bot.ensure_user(data['user'])
+    end
+
+    # Sets the server's AFK channel.
```
Comment (meew0) [inline lib/discordrb/data.rb]: AFK channel...?

File: lib/discordrb/data.rb [hunk 3]
```diff
@@ -269,6 +270,92 @@ def inspect
...
+    # @param afk_channel [Channel, nil] The new AFK channel, or `nil` if there should be none set.
+    def name=(name)
+      update_webhook_data(name: name)
+    end
+
+    # Sets the server's icon.
```
Comment (meew0) [inline lib/discordrb/data.rb]: Webhook's, not server's

File: lib/discordrb/data.rb [hunk 4]
```diff
@@ -269,6 +270,92 @@ def inspect
...
+        update_webhook_data(icon: icon)
+      end
+    end
+
+    # Deletes webhook.
+    def delete
+      API::Server.delete_webhook(@bot.token, @id)
+    end
+
+    # Utility function to get a webhook's avatar URL.
```
Comment (meew0) [inline lib/discordrb/data.rb]: I would call this "icon" rather than "avatar"

Comment (SnazzyPine25) [inline lib/discordrb/data.rb]: The API classifies it as a avatar.

**Comment by meew0** [comment]
Closing this in favour of #356 (which was spec'd and implemented the gateway event), sorry
